modules/client/src/internal/client/methods.ts

```diff
@@ -625,4 +637,75 @@ export class ClientMethods extends ClientCore implements IClientMethods {
       ),
     );
   }
+
+  /**
+   * Retrieves the list of plugins available on the PluginRegistry
+   *
+   * @param {PluginQueryParams} {
+   *     limit = 10,
+   *     skip = 0,
+   *     direction = SortDirection.ASC,
+   *     sortBy = PluginSortBy.SUBDOMAIN,
+   *     subdomain
+   *   }
+   * @return {*}  {(Promise<PluginRepo[] | null>)}
+   * @memberof ClientMethods
+   */
+  public async getPlugins({
+    limit = 10,
+    skip = 0,
+    direction = SortDirection.ASC,
+    sortBy = PluginSortBy.SUBDOMAIN,
+    subdomain,
+  }: PluginQueryParams = {}): Promise<PluginRepo[]> {
+    let where = {};
+    if (subdomain) {
+      where = { subdomain_contains_nocase: subdomain };
+    }
+    const query = QueryPlugins;
+    const params = {
+      where,
+      limit,
+      skip,
+      direction,
+      sortBy,
+    };
+    const name = "plugin repos";
```

Comment:
why this value and this variable?

Reply:
iss for the graphql error, by default the graphql request returns a generic error, here we just pass what caused the error